src/Extensions/SeoAICMSPageEditControllerExtension.php

```diff
@@ -49,29 +50,34 @@ public function generatePrompt()
         // Strip the content of header, footer and nav elements
         $domParser = HtmlDomParser::str_get_html(file_get_contents($pageLink));
 
-        foreach ($domParser->find('header') as $node) {
-            if ($node) {
-                $node->outertext = '';
-            }
+        $excludedDomElements = ['header', 'footer', 'nav'];
+        if($this->excluded_dom_selectors){
+            $excludedDomElements = $this->excluded_dom_selectors;
         }
 
-        foreach ($domParser->find('footer') as $node) {
-            if ($node) {
-                $node->outertext = '';
-            }
-        }
-        
-        foreach ($domParser->find('nav') as $node) {
-            if ($node) {
-                $node->outertext = '';
+        foreach ($excludedDomElements as $element) {
```

Comment:
Once we have added the properties, we can simplify this down to something like:

        $excludedDomElements = $this->excluded_dom_selectors;
        foreach ($excludedDomElements as $element) {
            foreach ($domParser->find($element) as $node) {
                if ($node) {
                    $node->outertext = '';
                }
            }
        }